tests/test_data_types.py

```diff
@@ -70,8 +65,15 @@ def test_no_data():
     data_type = data_type_factory("float32", no_data=math.nan)
     assert math.isnan(data_type.no_data)
 
-    with pytest.raises(ValueError):
-        data_type_factory("float32", no_data=1.1)
+    data_type = data_type_factory("float32", no_data=0.0)
+    assert data_type.no_data == 0.0
+
+    # Note: Surely this test is wrong? Ask Thomas.
+    # with pytest.raises(ValueError):
+    #     data_type_factory("float32", no_data=1.1)
+    # Surely this should be the correct behavior
```

Comment:
Please add test for `data_type_factory("float32", no_data=math.nan)` 
I don't spot the part where the incoming json value for `NaN` is converted into a float. I assume this happens in LayerModel. Maybe it is wise to add a test, just to make sure this works

```python
layer = LayerModel(..., no_data="nan")
assert isinstance(layer.no_data, float)
```